### PR TITLE
Test rewrites

### PR DESCRIPTION
**Pull request by liufengyun**
Test rewrites to avoid accidental regression

Related: #9002 #8982

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/parsing/Scanners.scala [hunk 1]
```diff
@@ -197,7 +197,6 @@ object Scanners {
     val noindentSyntax =
       ctx.settings.noindent.value
       || ctx.settings.oldSyntax.value
-      || migrateTo3
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/parsing/Scanners.scala:200]: I think having `-source 3.0-migrate` imply `-noindent` is intentional: it should allow existing weirdly formatted scala 2 code to compile /cc @odersky.

Comment (liufengyun) [reply compiler/src/dotty/tools/dotc/parsing/Scanners.scala:200]: The issue is that if a user run compiler with `-rewrite`, then it's impossible to compile the code again.

It's discovered when we actually compile the rewritten code in the following test:

```Scala
compileFile("tests/rewrites/rewrites.scala", scala2CompatMode.and("-rewrite", "-indent"))
```

Comment (smarter) [reply compiler/src/dotty/tools/dotc/parsing/Scanners.scala:200]: In that example I think `-indent` needs to take precedence other `migrateTo3` and force indentation to be on.

**Comment by liufengyun** [comment]
@smarter Could you have a look at the changes again?

**Pull request merged**